device/channels.go

```diff
@@ -111,7 +114,7 @@ type autodrainingOutboundQueue struct {
 // All sends to the channel must be best-effort, because there may be no receivers.
 func newAutodrainingOutboundQueue(device *Device) *autodrainingOutboundQueue {
```

Comment:
No complaints about this, but if you guys have any ideas how to reduce code duplication in this function through generics, maybe that'd be good. There's the issue of dispatch-based-on-type though for cleanup, and I dunno if Go's generics have this sort of C++-like ability yet. Just food for thought.

Reply:
I don't currently see an obvious way to make the finalizer funcs generic with type parameters.